### PR TITLE
feat(testing-framework): Testing framework refactoring

### PR DESCRIPTION
**Pull request by monoid**
1. Multiple AIR scripts can be executed within same network.

### CODE REVIEW

File: crates/testing-framework/src/execution/mod.rs [hunk 1]
```diff
@@ -15,23 +15,41 @@
...
 
-use std::{borrow::Borrow, collections::HashMap, hash::Hash, rc::Rc, str::FromStr};
+use std::{borrow::Borrow, hash::Hash, rc::Rc};
 
 pub struct TestExecutor {
-    pub air_script: String,
-    pub network: Network,
-    pub test_parameters: TestRunParameters,
+    transformee: Transformee,
+    test_parameters: TestRunParameters,
```
Comment (mikevoronov) [inline crates/testing-framework/src/execution/mod.rs:30]: `TestRunParameters` is not a property of a test network, but a property of particular script launch, so it should be passed only on script running

File: crates/testing-framework/src/ephemeral/mod.rs [hunk 2]
```diff
@@ -17,38 +17,35 @@
...
+use std::{borrow::Borrow, cell::RefCell, collections::HashMap, hash::Hash, rc::Rc};
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub struct PeerId(String);
+pub struct PeerId(Rc<str>);
 
 impl PeerId {
-    pub fn new(peer_id: impl Into<String>) -> Self {
-        Self(peer_id.into())
+    pub fn new<'a>(peer_id: impl Into<&'a str>) -> Self {
```
[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/ephemeral/mod.rs:36]: Ok, I use `'any`.

File: crates/testing-framework/src/queue.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use air_test_utils::{test_runner::TestRunParameters, RawAVMOutcome};
+
+use crate::ephemeral::{Data, Network, PeerId};
```
[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/queue.rs:19]: There is `fmt` option for this, but it either leaves as is or orders in std ordering which is opposite to ours.

File: crates/testing-framework/src/execution/mod.rs (hunk 1 above)
Comment (monoid) [reply crates/testing-framework/src/execution/mod.rs:30]: You may always create many test executors with different test_parameters, the behavior will be exactly the same, though not as explicit.

This particular design prevents using data from one particle with different particle, as the `queue` is bundled with the `test_parameters`.

**Review by mikevoronov — APPROVED** [review]
utACK

File: crates/testing-framework/src/asserts/mod.rs [hunk 4]
```diff
@@ -18,10 +18,14 @@ pub(crate) mod parser;
 
 use crate::services::JValue;
 
-use air_test_utils::CallServiceResult;
+use air_test_utils::{
+    prelude::{echo_call_service, unit_call_service},
+    CallRequestParams, CallServiceResult,
+};
+use serde_json::json;
```
[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/asserts/mod.rs:25]: I think it already comply to the ordering (crate/super newline other-non-std newline std).

File: crates/testing-framework/src/execution/mod.rs [hunk 5]
```diff
@@ -15,23 +15,41 @@
...
+    services::MarineServiceHandle,
+    transform::walker::Transformee,
 };
 
 use air_test_utils::{test_runner::TestRunParameters, RawAVMOutcome};
 
-use std::{borrow::Borrow, collections::HashMap, hash::Hash, rc::Rc, str::FromStr};
+use std::{borrow::Borrow, hash::Hash, rc::Rc};
 
 pub struct TestExecutor {
```
[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/execution/mod.rs:28]: Accepted.

[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/execution/mod.rs]: `ParticleExecutor`.

**Pull request merged**